pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -695,6 +695,17 @@ func (l *LibvirtDomainManager) generateConverterContext(vmi *v1.VirtualMachineIn
 		}
 	}
 
+	var efiConf *converter.EFIConfiguration
+	if vmi.IsBootloaderEFI() {
+		secureBoot := vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot == nil || *vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot
+
+		var err error
+		efiConf, err = converter.DetectEFIConfig(runtime.GOARCH, l.ovmfPath, secureBoot)
```

Comment:
You can detect the efi setup upfront and only fail if the VM wants to be secure booted. There is no need to detect it multiple times. Just an optional suggestion.

Reply:
That's a good idea! working on a v2, will post shortly